android/Staccato_AN/app/src/main/java/com/on/staccato/presentation/common/PhotoAttachFragment.kt

```diff
@@ -188,8 +201,31 @@ class PhotoAttachFragment : BottomSheetDialogFragment(), PhotoAttachHandler {
         }
     }
 
-    private fun showPermissionSnackBar() {
-        showSettingSnackBar(R.string.snack_bar_require_photo_album_permission)
+    private fun handleCameraPermissionNotGranted() {
+        showSettingSnackBar(R.string.photo_require_camera_permission)
+    }
+
+    private fun handleGalleryPermissionNotGranted() {
+        if (VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            val intent =
+                Intent(MediaStore.ACTION_PICK_IMAGES)
+                    .setDataAndType(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, "image/*")
+                    .setImageCountLimit()
+            galleryLauncher.launch(intent)
+        } else {
+            showSettingSnackBar(R.string.photo_require_photo_album_permission)
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.R)
+    private fun Intent.setImageCountLimit(): Intent {
+        if (!multipleAbleOption && SdkExtensions.getExtensionVersion(Build.VERSION_CODES.R) < 2) return this
+        availableImageCount?.let { imageCount ->
+            if (MIN_COUNT_FOR_PICK_IMAGES_MAX < imageCount) {
+                putExtra(MediaStore.EXTRA_PICK_IMAGES_MAX, imageCount)
+            }
+        }
```

Comment:
```kotlin
private var availableImageCount: Int? = null
```

availableImageCount의 초기값을 null보다는 0으로 설정하는 건 어떠신가요? 0으로 설정하면 null 체크를 할 필요가 없어서 depth를 줄일 수 있을 것 같아요!

```suggestion
        if (!multipleAbleOption && SdkExtensions.getExtensionVersion(Build.VERSION_CODES.R) < 2) return this
        if (MIN_COUNT_FOR_PICK_IMAGES_MAX < availableImageCount) {
            putExtra(MediaStore.EXTRA_PICK_IMAGES_MAX, availableImageCount)
        }
```

Reply:
**초기값**이면서도 **유효하지 않은 값**임이 더 명확하게 드러나도록 -1로 설정해주었습니다!